apps/builder/app/builder/features/ai/apply-operations.ts

```diff
@@ -0,0 +1,186 @@
+import { serverSyncStore } from "~/shared/sync";
+import { generateDataFromEmbedTemplate } from "@webstudio-is/react-sdk";
+import { copywriter, operations } from "@webstudio-is/ai";
+import { isBaseBreakpoint } from "~/shared/breakpoints";
+import {
+  deleteInstance as _deleteInstance,
+  insertTemplateData,
+} from "~/shared/instance-utils";
+import {
+  breakpointsStore,
+  instancesStore,
+  registeredComponentMetasStore,
+  selectedInstanceStore,
+  styleSourceSelectionsStore,
+  styleSourcesStore,
+  stylesStore,
+} from "~/shared/nano-states";
+import type { DroppableTarget } from "~/shared/tree-utils";
+import { getStyleDeclKey, type StyleSource } from "@webstudio-is/sdk";
+import { nanoid } from "nanoid";
+import { traverseTemplate } from "@webstudio-is/jsx-utils";
+
+export const applyOperations = (operations: operations.Response) => {
+  for (const operation of operations) {
+    switch (operation.operation) {
+      case "insertTemplate":
+        traverseTemplate(operation.template, (node) => {
+          if (node.type === "instance" && node.component.startsWith("Radix.")) {
+            node.component =
+              "@webstudio-is/sdk-components-react-radix:" +
+              node.component.slice("Radix.".length);
+          }
+        });
+        insertTemplateByOp(operation);
+        break;
+      case "deleteInstance":
+        deleteInstanceByOp(operation);
+        break;
+      case "applyStyles":
+        applyStylesByOp(operation);
+        break;
+      default:
+        if (process.env.NODE_ENV === "development") {
+          // eslint-disable-next-line no-console
+          console.warn(`Not supported operation: ${operation}`);
+        }
+    }
+  }
+};
+
+const insertTemplateByOp = (
+  operation: operations.generateInsertTemplate.wsOperation
+) => {
+  const breakpoints = breakpointsStore.get();
+  const breakpointValues = Array.from(breakpoints.values());
+  const baseBreakpoint = breakpointValues.find(isBaseBreakpoint);
+  if (baseBreakpoint === undefined) {
+    return false;
+  }
+  const metas = registeredComponentMetasStore.get();
+  const templateData = generateDataFromEmbedTemplate(
+    operation.template,
+    metas,
+    baseBreakpoint.id
+  );
+
+  // Occasionally the LLM picks a component name as the insertion point.
+  // Instead of throwing the otherwise correct operation we try to fix this here.
+  if (
+    [...metas.keys()].some((ComponentName) =>
+      ComponentName.includes(operation.addTo)
+    )
```

Comment:
Could you clarify why you check substring in component name?
And please start it with lower case